src/main/java/com/csc4003/apis/models/Booking.java

```diff
@@ -62,50 +52,136 @@ public void setBookingId(int bookingId) {
         this.bookingId = bookingId;
     }
 
-    public int getEmployeeId() {
-        return employeeId;
+    public Timestamp getStartTime() {
+        return startTime;
     }
 
-    public void setEmployeeId(int employeeId) {
-        this.employeeId = employeeId;
+    public void setStartTime(Timestamp startTime) {
+        this.startTime = startTime;
     }
 
-    public Integer getRoomId() {
-        return roomId;
+    public int getDuration() {
+        return duration;
     }
 
-    public void setRoomId(int roomId) {
-        this.roomId = roomId;
+    public void setDuration(int duration) {
+        this.duration = duration;
     }
 
-    public Integer getDeskId() {
-        return deskId;
+    public Room getRoom() {
+        return room;
     }
 
-    public void setDeskId(int deskId) {
-        this.deskId = deskId;
+    public Desk getDesk() {
+        return desk;
     }
 
-    public Timestamp getStartTime() {
-        return startTime;
+    public Employee getEmployee() {
+        return employee;
     }
 
-    public void setStartTime(Timestamp startTime) {
-        this.startTime = startTime;
+    public void setRoom(Room room) {
+        this.room = room;
     }
 
-    public int getDuration() {
-        return duration;
+    public void setDesk(Desk desk) {
+        this.desk = desk;
     }
 
-    public void setDuration(int duration) {
-        this.duration = duration;
+    public void setEmployee(Employee employee) {
+        this.employee = employee;
     }
 
     @Override
     public String toString() {
-        return "Booking: {" + "bookingId=" + bookingId +  ", employeeId=" + employeeId + ", roomId=" + roomId
-                +  ", deskId=" + deskId +  ", startTime=" + startTime + ", duration=" + duration +'}';
+        return "Booking: {" + "bookingId=" + bookingId +  ", employeeId=" + employee.getEmployeeId() + ", roomId=" + room.getRoomId()
+                +  ", deskId=" + desk.getDeskId() +  ", startTime=" + startTime + ", duration=" + duration +'}';
+    }
+
+    // ---- Room Details Methods
+    public String getRoomNumber() {
+        return room.getRoomNumber();
```

Comment:
I think there should be no worry about this, the API should check if it's a room or desk booking that's being asked for and return the details accordingly avoiding the error. If not I can change this in future to avoid this issue coming up.